docs/design/da-pki-revocation-design.md

```diff
@@ -0,0 +1,213 @@
+   If `CRLSignerCertificate` is a PAI, then it must be chained back to a valid PAA present on the ledger.
+
+3. If `IssuerSubjectKeyID` is not equal to the `Subject Key Identifier` of `CRLSignerCertificate`, the transaction is considered invalid and rejected.
+
+4. If `RevocationType` is 1 (RFC5280 CRL), then `DataFileSize`, `DataDigest`, `DataDigestType` must be empty.
+
+   Otherwise, it may cause confusion. Certificates to be revoked are added to the content of CRL URL without a need to send a transaction to the ledger.
+   So, someone may look at the DCL entry, see that a file size is specified, then go to the URL, and see that the actual size there is different.
+   It may cause a question if the data is valid, and why a different file size is in DCL.
+
+5. `ProductID` is expected to be `unit16`, not `unit32` as in other places.
+
+
+## A need for Proof-of-possession of `CRLSignerCertificate` key
+No additional proof-of-possession is required because 
+- The transaction is already signed by the Vendor account key, so only a trusted Vendor can send it.
+- DCL checks that `CRLSignerCertificate` is either a valid PAI (signed by a trusted PAA from DCL) or a PAA present on the ledger.
+- It can be complicated to create an interactive PoP of `CRLSignerCertificate` for some PAIs.
+- Indirect CRLs are out of scope for now
+
+## Transactions
+
+### 1. ADD_PKI_REVOCATION_DISTRIBUTION_POINT
+Publishes a PKI Revocation distribution endpoint owned by the Vendor. 
+
+If `crlSignerCertificate` is a PAA (root certificate), then it must be present on DCL.
+
+If `crlSignerCertificate` is a PAI (intermediate certificate), then it must be chained back to a valid PAA (root certificate) present on DCL.
+In this case `crlSignerCertificate` is not required to be present on DCL, and will not be added to DCL as a result of this transaction.
+If PAI needs to be added to DCL, it should be done via `ADD_X509_CERT` transaction.
+
+Publishing the revocation distribution endpoint doesn't automatically remove PAI (Intermediate certificates)
+and DACs (leaf certificates) added to DCL if they are revoked in the CRL identified by this distribution point.
+
+
+- Who can send:
+    - Vendor account
+    - `vid` field in the transaction (`VendorID`) must be equal to the Vendor account's VID
+    - `vid` field in the `CRLSignerCertificate` (for vendor-scoped PAAs and PAIs) must be equal to the Vendor account's VID
+- Parameters:
+    - vid: `uint16` -  Vendor ID (positive non-zero)
+    - pid: `optional(uint16)` -  Product ID (positive non-zero)
+    - isPAA: `bool` -  True if the revocation information distribution point relates to a PAA
+    - label: `string` -  A label to disambiguate multiple revocation information partitions of a particular issuer.
+    - crlSignerCertificate: `string` -  PEM encoded certificate (string or path to file containing data)
```

Comment:
Always the actual certificate. Not a path to file containing data.

Reply:
Yes, the actual transaction will always consist the actual certificate. We are going to support path for the corresponding parameter in CLI (as we do with other X509 related CLI tins). I will update it to avoid confusion.